### ISSUE TITLE
1.2.0-hadoop2.7.4-java8 : NodeManager doesn't satisfy minimum allocations

### ISSUE THREAD

**Issue by JMLamodiere** [big-data-europe/docker-hadoop#16]
When switching all images version in [docker-compose.yml](https://github.com/big-data-europe/docker-hadoop/blob/master/docker-compose.yml) from `1.1.0-hadoop2.7.1-java8` to `1.2.0-hadoop2.7.4-java8` , my `nodemanager1` exit with code `255`. The logs (`docker-compose logs nodemanager1`) contain this error : 

```
nodemanager1       | 17/10/04 12:56:21 ERROR nodemanager.NodeStatusUpdaterImpl: Unexpected error starting NodeStatusUpdater
nodemanager1       | org.apache.hadoop.yarn.exceptions.YarnRuntimeException: Recieved SHUTDOWN signal from Resourcemanager ,Registration of NodeManager failed, Message from ResourceManager: NodeManager from  ba78946515dc doesn't satisfy minimum allocations, Sending SHUTDOWN signal to the NodeManager.
nodemanager1       | 	at org.apache.hadoop.yarn.server.nodemanager.NodeStatusUpdaterImpl.registerWithRM(NodeStatusUpdaterImpl.java:278)
nodemanager1       | 	at org.apache.hadoop.yarn.server.nodemanager.NodeStatusUpdaterImpl.serviceStart(NodeStatusUpdaterImpl.java:197)
nodemanager1       | 	at org.apache.hadoop.service.AbstractService.start(AbstractService.java:193)
nodemanager1       | 	at org.apache.hadoop.service.CompositeService.serviceStart(CompositeService.java:120)
nodemanager1       | 	at org.apache.hadoop.yarn.server.nodemanager.NodeManager.serviceStart(NodeManager.java:272)
nodemanager1       | 	at org.apache.hadoop.service.AbstractService.start(AbstractService.java:193)
nodemanager1       | 	at org.apache.hadoop.yarn.server.nodemanager.NodeManager.initAndStartNodeManager(NodeManager.java:496)
nodemanager1       | 	at org.apache.hadoop.yarn.server.nodemanager.NodeManager.main(NodeManager.java:543)
```

**TL;DR** : `Message from ResourceManager: NodeManager from  ba78946515dc doesn't satisfy minimum allocations, Sending SHUTDOWN signal to the NodeManager`

It works fine with `1.1.0-hadoop2.7.1-java8`. I can't figure out what change intruduced by version `1.2.0-hadoop2.7.4-java8` leads to this error. Am I the only one getting the error ?

FYI all `docker-compose.yml` in every git branches always refers to `1.1.0-hadoop2.7.1-java8`, including branches `master` and `1.2.0-hadoop2.7.4-java8`.

**Comment by earthquakesan** [comment]
@JMLamodiere,

> FYI all docker-compose.yml in every git branches always refers to 1.1.0-hadoop2.7.1-java8, including branches master and 1.2.0-hadoop2.7.4-java8.
thanks for pointing out, I will fix it

Regarding minimum allocation, you need to define node manager memory and cpus explicitly, for some reason in version 2.7.4 hadoop inside docker got wrong info on available resources.
What you need to do is to add this as env vars:
```
YARN_CONF_yarn_nodemanager_resource_memory___mb=16384
YARN_CONF_yarn_nodemanager_resource_cpu___vcores=8
```

For working example, see the [hadoop stack](https://github.com/big-data-europe/docker-hbase/blob/master/docker-compose-hadoop-yarn.yml) I've been using for hbase deployment. In particular, [env file](https://github.com/big-data-europe/docker-hbase/blob/master/hadoop.env).

**Comment by earthquakesan** [comment]
@JMLamodiere hi!

FYI, the problem occurs because Java version we use currently can not read resource allocation from cgroup. The problem is described [here](https://developers.redhat.com/blog/2017/03/14/java-inside-docker/). The solution is to use openjdk java8u131+

**Comment by JMLamodiere** [comment]
Hi !
Ok thanks ! Your solution above with 2 ENV vars did fix my problem, but I'll give a chance to openjdk.

**Comment by earthquakesan** [comment]
@JMLamodiere sorry, but don't waste your time. I tried it and it throws the same error (i.e. nodemanager does not see available resources on a node). The further improvement would be automatic injection of env vars using script which can read cgroup restrictions. 

At the moment, the first solution is one to go. Cheers.

**Comment by gkpavan001** [comment]
Hi,
I am getting the below error while running hadoop 2.7.4 through windows
**"resourcemanager.ResourceTrackerService: NodeManager from  DESKTOP-EJ1B1V9 doesn't satisfy minimum allocations, Sending SHUTDOWN signal to the NodeManager."**

In which file need to add the below properties?
YARN_CONF_yarn_nodemanager_resource_memory___mb=16384
YARN_CONF_yarn_nodemanager_resource_cpu___vcores=8
